### ISSUE TITLE
Swift Package Manager hangs on xcodebuild

### ISSUE THREAD

**Issue by grzegorzkrukowski** [actions/runner-images#6233]
### Description

Our builds started to freeze on the SPM resolving package graph during xcodebuild phase.

We tried disabling caching for SPM, but it didn't matter.
It randomly stops on checking out some SPMs (randomly), for example:
`Checking out x.x.x of package ‘Dependency’`

We tried with Xcode 13.4.1 and 14.0 - both are going through SPM package resolving.
We increased the timeout to 50 minutes, but it didn't help.

We also re-run our previously succeeding workflow, and they are failing today on precisely the same code, settings, and configuration.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [X] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Version: 20220907.1
https://github.com/actions/runner-images/releases/tag/macOS-12%2F20220907.1

### Is it regression?

yes

### Expected behavior

SPM should resolve dependencies without hanging on checkouts

### Actual behavior

SPM is hanging on resolving dependencies

### Repro steps

1. Prepare an xcode project with SPM dependencies
2. Try running xcodebuild to resolve SPM dependencies

**Comment by mikhailkoliada** [comment]
@grzegorzkrukowski Hi! Could you provide a minimalistic pipeline steps to reproduce the issue?

**Comment by grzegorzkrukowski** [comment]
@mikhailkoliada I will try to extract the minimalistic project and workflow that will reproduce the issue. It is happening for our private repository which is not easy to share unfortunately.

**Comment by grzegorzkrukowski** [comment]
Hey. While trying to prepare a public project to present the problem, we nailed down the issue.

It seems that SPM starting from Xcode 13.3, with .netrc authorization support, has a weird behavior when it's asking for Keychain access even for public repositories declared with https.
More in this thread:
https://forums.swift.org/t/spm-support-basic-auth-for-non-git-binary-dependency-hosts/37878

This was causing our pipeline to freeze on CI (there are no logs in that case on runners).

We resolved the problem, by settings up .netrc on our machines, and things are working again.
